core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -53,6 +53,8 @@ object RequestChannel extends Logging {
     @volatile var apiLocalCompleteTimeMs = -1L
     @volatile var responseCompleteTimeMs = -1L
     @volatile var responseDequeueTimeMs = -1L
+    @volatile var apiRemoteCompleteTimeMs = -1L
```

Comment:
Filed this to add a test. https://issues.apache.org/jira/browse/KAFKA-2699 

I think apiLocalCompleteTimeMs can't be zero. The < 0 check exists to identify when the value has not been set.